core/templates/services/literally-canvas-helper.service.ts

```diff
@@ -58,31 +58,240 @@ angular.module('oppia').factory('LiterallyCanvasHelperService', [
+          text += a.textContent + '\n';
+        });
+        shape.data.text = text.slice(0, -1);
+        shape.data.color = node.attributes.fill.value;
+        shape.data.font = node.attributes.style.value.slice(6, -1);
+        shape.data.forcedWidth = (
+          typeof node.attributes.width !== 'undefined' ? (
+            parseFloat(node.attributes.width.value)) : 0);
+        shape.data.forcedHeight = (
+          typeof node.attributes.height !== 'undefined' ? (
+            parseFloat(node.attributes.height.value)) : 0);
+        shape.id = id.slice(1).join('-');
+      }
+      return shape;
+    };
+
     return {
-      isSVGTagValid: function(svgString) {
+      svgParse: function(svgString, lc) {
+        // This function is used to convert svg to snapshot object which is
+        // used by literallyCanvas to load the svg into canvas.
         var domParser = new DOMParser();
-        var doc = domParser.parseFromString(svgString, 'image/svg+xml');
-        var allowedTags = Object.keys(CONSTANTS.SVG_ATTRS_WHITELIST);
-        var nodeTagName = null;
-        var valid = true;
-        doc.querySelectorAll('*').forEach((node) => {
-          nodeTagName = node.tagName.toLowerCase();
-          if (allowedTags.indexOf(nodeTagName) !== -1) {
-            for (var i = 0; i < node.attributes.length; i++) {
-              if (CONSTANTS.SVG_ATTRS_WHITELIST[nodeTagName].indexOf(
-                node.attributes[i].name.toLowerCase()) === -1) {
-                valid = false;
-              }
-            }
-          } else {
-            valid = false;
+        var doc = domParser.parseFromString(svgString, 'text/xml');
+        var snapshot = {
+          colors: {
+            primary: '',
+            secondary: '',
+            background: ''
+          },
+          position: '',
+          scale: 1,
+          shapes: [],
+          backgroundShapes: [],
+          imageSize: {
+            width: '',
+            height: ''
           }
+        };
+        var rect = doc.querySelector('svg > rect');
+        snapshot.colors.primary = lc.colors.primary;
+        snapshot.colors.secondary = lc.colors.secondary;
+        // @ts-ignore fill property doesnot exist error.
+        snapshot.colors.background = rect.attributes.fill.value;
+        snapshot.position = lc.position;
+        snapshot.backgroundShapes = lc.backgroundShapes;
+        // @ts-ignore width property doesnot exist error.
+        snapshot.imageSize.width = rect.attributes.width.value;
+        // @ts-ignore height property doesnot exist error.
+        snapshot.imageSize.height = rect.attributes.height.value;
+
+        doc.querySelectorAll('svg > g > *').forEach((node) => {
+          snapshot.shapes.push(svgToShapeObject(node));
         });
-        if (!valid) {
-          throw new Error('Invalid tag or attribute in svg.');
-        }
-        return valid;
+        return snapshot;
       },
+
+      svgTagIsValid: function(svgString) {
```

Comment:
isSvgTagValid because this is a function. 

Also, do we need this? Can't we directly call validateSVGTag instead?

Reply:
We can't use validateSVGTag directly here because if I return validateSVGTag then I have to use "this" keyword to refer to it and when the svgrenderer functions are used by literally canvas it causes a problem with "this".

Reply:
Makes sense. Thanks for explaining!